mongo/integration/unified/collection_operation_execution_test.go

```diff
@@ -524,6 +524,122 @@ func executeFind(ctx context.Context, operation *Operation) (*OperationResult, e
 	return NewCursorResult(docs), nil
 }
 
+func executeFindOneAndDelete(ctx context.Context, operation *Operation) (*OperationResult, error) {
+	coll, err := Entities(ctx).Collection(operation.Object)
+	if err != nil {
+		return nil, err
+	}
+
+	var filter bson.Raw
+	opts := options.FindOneAndDelete()
+
+	elems, _ := operation.Arguments.Elements()
+	for _, elem := range elems {
+		key := elem.Key()
+		val := elem.Value()
+
+		switch key {
+		case "collation":
+			collation, err := createCollation(val.Document())
+			if err != nil {
+				return nil, fmt.Errorf("error creating collation: %v", err)
+			}
+			opts.SetCollation(collation)
+		case "filter":
+			filter = val.Document()
+		case "hint":
+			hint, err := createHint(val)
+			if err != nil {
+				return nil, fmt.Errorf("error creating hint: %v", err)
+			}
+			opts.SetHint(hint)
+		case "maxTimeMS":
+			opts.SetMaxTime(time.Duration(val.Int32()) * time.Millisecond)
+		case "projection":
+			opts.SetProjection(val.Document())
+		case "sort":
+			opts.SetSort(val.Document())
+		default:
+			return nil, fmt.Errorf("unrecognized findOneAndDelete option %q", key)
+		}
+	}
+	if filter == nil {
+		return nil, newMissingArgumentError("filter")
+	}
+
+	res, err := coll.FindOneAndDelete(ctx, filter, opts).DecodeBytes()
+	return NewDocumentResult(res, err), nil
+}
+
+func executeFindOneAndReplace(ctx context.Context, operation *Operation) (*OperationResult, error) {
+	coll, err := Entities(ctx).Collection(operation.Object)
+	if err != nil {
+		return nil, err
+	}
+
+	var filter bson.Raw
+	var replacement interface{}
+	opts := options.FindOneAndReplace()
+
+	elems, _ := operation.Arguments.Elements()
+	for _, elem := range elems {
+		key := elem.Key()
+		val := elem.Value()
+
+		switch key {
+		case "bypassDocumentValidation":
+			opts.SetBypassDocumentValidation(val.Boolean())
+		case "collation":
+			collation, err := createCollation(val.Document())
+			if err != nil {
+				return nil, fmt.Errorf("error creating collation: %v", err)
+			}
+			opts.SetCollation(collation)
+		case "filter":
+			filter = val.Document()
+		case "hint":
+			hint, err := createHint(val)
+			if err != nil {
+				return nil, fmt.Errorf("error creating hint: %v", err)
+			}
+			opts.SetHint(hint)
+		case "maxTimeMS":
+			opts.SetMaxTime(time.Duration(val.Int32()) * time.Millisecond)
+		case "projection":
+			opts.SetProjection(val.Document())
+		case "replacement":
+			replacement, err = createUpdateValue(val)
```

Comment:
The replacement should always be a document. We use `createUpdateValue` for `FindOneAndUpdate` because the `update` parameter can either be a document or an array for pipeline-style updates in 4.4+. You can change this to declare `replacement` as a `bson.Raw` and treat it the same way as `filter` for this function.

Reply:
Gotcha, switched to just `replacement = val.Document()` and a `bson.Raw`.